pkg/rpcutil/server.go

```diff
@@ -139,7 +160,7 @@ func (h PreRPCHooker[T]) forwardToLeader(
 	}
 	if needForward {
 		h.leaderCli.RLock()
```

Comment:
will make RWMutex a private member in next commit, to avoid this silly mistake 😂